### PR TITLE
Rework documentation of std.format.package : sformat

### PR DESCRIPTION
**Pull request by berni44**
I removed `buf` from the first line, because this line might be used in other contexts, where `buf` is not known (for example in overview tables).

I switched the order of the two overloads, because in my opinion, the overload that resembles a function call in other languages (and without template constraints) is better understandable by new users and with the switch it's shown first in the docs. This is also the order of the examples below: First "normal" use, then check at compile time.

I removed the reference to `format()` in the comment about checks at compile time, because I couldn't find anything useful there, apart from a note, that it might produce faster code, which only applies to `format` but not to `sformat`.

I replaced the example, because I dislike unimaginative foo-bar examples and I wanted to put more emphasize on how the function has to be called, when checking at compile time is wished for.

### CODE REVIEW

File: std/format/package.d [hunk 1]
```diff
@@ -1853,29 +1853,39 @@ unittest
...
-    alias e = checkFormatException!(fmt, Args);
-    static assert(!e, e.msg);
-    return .sformat(buf, fmt, args);
-}
+/**
+Converts its arguments according to a format string into a buffer.
+The buffer has to be large enough to hold the formatted string.
 
-/// ditto
+The second version of `sformat` takes the format string as template
```
Comment (RazvanN7) [inline std/format/package.d:1860]:
```suggestion
The second version of `sformat` takes the format string as a template
```

File: std/format/package.d [hunk 2]
```diff
@@ -1853,29 +1853,39 @@ unittest
...
+argument. In this case, it is checked for consistency at
+compile-time.
+
+Params:
+    buf = the buffer, where the formatted result should go
```
Comment (RazvanN7) [inline std/format/package.d:1865]:
```suggestion
    buf = the buffer where the formatted string should go
```

File: std/format/package.d [hunk 3]
```diff
@@ -1853,29 +1853,39 @@ unittest
...
+    A slice of `buf` containing the formatted string.
+
+Throws:
+    A $(REF_ALTTEXT RangeError, RangeError, core, exception) if `buf`
+    isn't large enough to hold the formatted string
+    and a $(LREF FormatException) if formatting did not succeed.
+
+Note:
+    In theory this function should be `@nogc`. But with the current
+    implementation there are some cases, where allocations occur:
```
Comment (RazvanN7) [inline std/format/package.d:1881]:
```suggestion
    implementation there are some cases where allocations occur:
```

File: std/format/package.d [hunk 4]
```diff
@@ -1931,13 +1941,43 @@ char[] sformat(Char, Args...)(return scope char[] buf, scope const(Char)[] fmt,
...
+
+/// The format string can be checked at compile-time:
+@safe pure unittest
+{
+    char[20] buf;
+
+    assert(sformat!"Here are %d %s."(buf[], 3, "apples") == "Here are 3 apples.");
+
+    // This line doesn't compile, because 3.14 cannot be formatted with %d:
+    // writeln(sformat!"Here are %d %s."(buf[], 3.14, "apples"));
```
Comment (RazvanN7) [inline std/format/package.d:1970]: It looks kind of bad to have commented code in public examples. Will the unittest become `@system` if we replace this comment with: `static assert(!__traits(compiles, sformat!"Here are %d %s."(buf[], 3.14, "apples")));` ?

**Review by RazvanN7 — CHANGES_REQUESTED** [review]

File: std/format/package.d (hunk 4 above)
Comment (berni44) [reply std/format/package.d:1970]: Using `__traits` is possible, but I dislike it. For a beginner, I think, it's much more comprehensible to have commented code, than a `__traits` statement. (Note, that I added this `__traits` statement in the next (nonpublic) unittest, to actually verify this statement.)

Whatever we decide upon here, we should do the same for the similar example of `format` (a PR that was already merged yesterday).

**Review by RazvanN7 — APPROVED** [review]

File: std/format/package.d (hunk 4 above)
Comment (RazvanN7) [reply std/format/package.d:1970]: Fair enough.

**Pull request merged**